assert/equal.ts

```diff
@@ -101,16 +96,13 @@ export function equal(c: unknown, d: unknown): boolean {
         if (primitiveKeysFastPath) {
           if (a instanceof Set) {
             return a.symmetricDifference(b).size === 0;
+          } else if (b instanceof Set) {
+            // Already checked constructorsEqual
+            throw new Error("Unreachable");
```

Comment:
Can you please remove this `else if`?

Reply:
It's necessary to narrow the type of `b`. What would be preferable? Just cast with `as`?

Reply:
Yes please. Let's cast with `as`.